docs/mitigating_ret2spec.md

```diff
@@ -0,0 +1,122 @@
+# Mitigating ret2spec
+
+This doc explains ret2spec attacks and how they are mitigated between different execution modes. In particular, we explain the subtleties of how the Linux kernel is protected from ret2spec attacks originating from user processes.
```

Comment:
(Do we have any tooling we can use here to autoformat md to short line lengths?)

Reply:
I'm sure I could use a better editor, or use my current editor more competently. But I don't know offhand of another tool for the job.

I do see that Google's style for internal markdown (g3doc) came down in favor of 80-char lines, though perhaps that's because we have corresponding tooling to make that easy. Anyway, sorry for the inconsistency.

When we fix this we'll need to fix https://raw.githubusercontent.com/google/safeside/main/docs/fencing.md too.